pkg/replica/replica.go

```diff
@@ -773,10 +773,22 @@ func (r *Replica) linkDisk(oldname, newname string) error {
 
 	dest := r.diskPath(newname)
 	if _, err := os.Stat(dest); err == nil {
-		logrus.Infof("Old file %s exists, deleting", dest)
+		destMetadata := r.diskPath(newname + metadataSuffix)
+		if _, err := os.Stat(destMetadata); err == nil {
+			logrus.Infof("Deleting old disk metadata file %s", destMetadata)
+			if err := os.Remove(destMetadata); err != nil {
```

Comment:
Just note: the disk meta should be the source of truth to ensure the atomicity of each disk (image + metafile).

https://github.com/longhorn/longhorn-engine/blob/c3a9bf904fa7b02737b93bd18efb24d5f32661c7/pkg/replica/replica.go#L187-L186

Reply:
Can we blindly remove the existing meta file then the image file here? I mean, using `os.RemoveAll(filePath)` without `os.State()`

Reply:
> Can we blindly remove the existing meta file then the image file here? I mean, using `os.RemoveAll(filePath)` without `os.State()`

Yes, we can replace `Stat...Remove` with `RemoveAll`.

Reply:
Updated. Thank you.